.github/workflows/prepare_release.yml

```diff
@@ -0,0 +1,53 @@
+name: Prepare release
+on:
+  push:
+    tags:
+      - "v*.*.*"
+
+jobs:
+  prepare_release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set RELEASE_VERSION
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      - name: Prepare release
+
+      - name: Setup Python 3.8
+        uses: actions/setup-python@v2
+        with: python-version:3.8
+
+      - name: Python Poetry Action
+        uses: abatilo/actions-poetry@v2.1.0
+
+      - name: Install dependencies
+        run: poetry install
+
+      - name: Get current version
+        run: echo "CURRENT_VERSION=$(poetry version | awk '{print $2}')" >> $GITHUB_ENV
+
+      - name: Generate changelog
+        run: |
+          poetry run tools/gen_changelog.py ${{ secrets.GITHUB_TOKEN }} v$CURRENT_VERSION v$RELEASE_VERSION
```

Comment:
This will fail because `gen_changelog.py` is not an executable; you need to `chmod +x tools/gen_changelog.py`.
Also, this needs `git` and `github` python modules (named `gitpython` and `pygithub`), so you need to add those to the dependencies.

Also, if your tags are like `vA.B.C` then I think the `v` in `v$RELEASE_VERSION` should be removed. You might also need to remove the extra from the pull request body as well.

Reply:
Thank you for your review. All your comments have been addressed. As for the `$RELEASE_VERSION` variable, it holds only the version without the `v`.